consensus/polybft/state_sync_manager.go

```diff
@@ -0,0 +1,483 @@
+package polybft
+
+import (
+	"encoding/hex"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"path"
+	"sync"
+
+	"github.com/0xPolygon/polygon-edge/consensus/polybft/bitmap"
+	polybftProto "github.com/0xPolygon/polygon-edge/consensus/polybft/proto"
+	bls "github.com/0xPolygon/polygon-edge/consensus/polybft/signer"
+	"github.com/0xPolygon/polygon-edge/consensus/polybft/wallet"
+	"github.com/0xPolygon/polygon-edge/tracker"
+	"github.com/0xPolygon/polygon-edge/types"
+	"github.com/hashicorp/go-hclog"
+	"github.com/libp2p/go-libp2p/core/peer"
+	"github.com/umbracle/ethgo"
+	"google.golang.org/protobuf/proto"
+)
+
+// StateSyncManager is an interface that defines functions for state sync workflow
+type StateSyncManager interface {
+	Init() error
+	Commitment() (*CommitmentMessageSigned, error)
+	PostBlock(req *PostBlockRequest) error
+	PostEpoch(req *PostEpochRequest) error
+}
+
+var _ StateSyncManager = (*dummyStateSyncManager)(nil)
+
+// dummyStateSyncManager is used when bridge is not enabled
+type dummyStateSyncManager struct{}
```

Comment:
Can this be `nil` instead of dummy?

Reply:
Fix: https://github.com/0xPolygon/polygon-edge/pull/1067/commits/7ad92ae33af92531583957b17d1a9ffdc8645521

Reply:
:D no, I mean if we can not assign `nil` value to the var? In case we can not, `dummy` is good name hahah

Reply:
Aaaahh, I see your point :smile: . By default (when bridge is not configured), we will always use this `nil(dummy)StateSyncManager` in the `runtime` constructor. So basically, it will always be either this dummy version, or the full state sync manager version.